### PR TITLE
Added forecastMove to game. When writing an ai agent it is useful to …

### PR DESCRIPTION
**Pull request by jacoblangston**
…create a deep copy of the board, then apply a possible move, and return an updated game board without modifying the actual board.

### CODE REVIEW

**Comment by jacoblangston** [comment]
Addresses https://github.com/arbrown/pao/issues/75

**Comment by arbrown** [comment]
I'll merge this in if you find it useful, but I think it might not serve the function you think it will for an AI for two reasons:

1. The boardstate representation using slices of strings is convenient for JSON serialization but very memory inefficient at the scale of a dense game tree like in ban qi.  I wrote a whole other separate library for representing the boardstate as a set of [bitboards](https://www.chessprogramming.org/Bitboards) which saves so much space.
1. Even then, I only keep one copy and just do/undo the move (which is similarly packed into a small int based data structure) while traversing the tree, because keeping a copy of the board state is still really inefficient, no matter how you pack it.

File: game/game.go [hunk 1]
```diff
@@ -562,6 +562,62 @@ func (g *Game) performMove(m *move) (bool, string) {
 	return true, tgtPiece
 }
 
+// Some logic is duplicated with the function performMove. Look into reusing logic where possible.
+func forecastMove(currentBoard [][]string, pieceToInt map[string]int, canAttack [][]bool, srcFile, srcRank, tgtFile, tgtRank int) (bool, [][]string) {
```
Comment (arbrown) [inline game/game.go:566]: If this function is going to be exported and accessible outside the package (eg, for anAI,) it needs to be capitalized. `ForecastMove`.

Comment (arbrown) [inline game/game.go:566]: Could you change this to use [named return values](https://tour.golang.org/basics/7)?  Feel free to update the other function for consistency.

**Comment by jacoblangston** [comment]
I used this same methodology for creating an adversarial AI for the game Isolation. The ForecastMove method is a base method in order to be able to build an adversarial AI. The reason it is useful is you can simulate future states of the game board, several moves in advance (usually limited by a time threshold). This functionality is needed to implement the minimax algorithm.

I'll make the requested changes.

**Pull request closed without merging**